### PR TITLE
[Fleet] Test JWT

### PR DESCRIPTION
**Pull request by nchaulet**
POC of auth with JWT

### CODE REVIEW

**Comment by nchaulet** [comment]
buildkite run perf-tests

File: internal/pkg/apikey/apikey.go [hunk 1]
```diff
@@ -124,6 +124,10 @@ func NewAPIKeyFromToken(token string) (*APIKey, error) {
 
 // Token returns the b64 encoded token of the APIKey.
 func (k APIKey) Token() string {
+	if strings.Contains(k.ID, "jwt:") {
```
[the remark this answers is not in the document]
Comment (nchaulet) [reply internal/pkg/apikey/apikey.go:127]: We will have to change the `authAPIKey` to support both JWT and ES api key to make this compatible

File: internal/pkg/api/auth.go [hunk 2]
```diff
@@ -35,6 +38,39 @@ func authAPIKey(r *http.Request, bulker bulk.Bulk, c cache.Cache) (*apikey.APIKe
...
 
+	//  Hacky way to have test passing
+	if !strings.Contains(r.URL.Path, "/enroll") && !testing.Testing() {
+		jwtToken, err := apikey.ExtractJWTAPIKey(r)
+		if err != nil {
+			return nil, err
+		}
+		token, err := jwt.Parse(jwtToken, func(token *jwt.Token) (interface{}, error) {
+			// Don't forget to validate the alg is what you expect:
+			if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
```
Comment (adamkasztenny) [inline internal/pkg/api/auth.go:49]: Is the token signature being validated with the private key that was used to sign the token?

Comment (adamkasztenny) [reply internal/pkg/api/auth.go:49]: Also, in addition to the algorithm, you'd want to check all of the [registered claim names](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1) (except `iat` and `jti`).

**Pull request closed without merging**